### PR TITLE
Show hide taskpane

### PR DESCRIPTION
**Pull request by Rick-Kirkham**

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[0ddc7e3](https://github.com/OfficeDev/office-js-docs-pr/commits/0ddc7e33ca1809fe3a9b263b6d6cce10bcb0b3b9)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C26%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002262339246263-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=snbQXgPKNIGT4KX7ztLxuwVAeo6J%2Bnz2DQ3rcYC5%2BY4%3D&st=2020-02-26T23%3A35%3A56Z&se=2020-03-28T23%3A40%3A56Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/show-hide-add-in.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+title: Show or hide an Office Add-in in a shared runtime
+description: 'Learn how to programmatically hide or show the UI of an add-in while it runs continuously'
+ms.date: 03/02/2020
+localization_priority: Normal
+---
+
+# Show or hide an Office Add-in in a shared runtime (preview)
+
+Office Add-ins can include any of the following parts:
```
Comment (mscharlock) [inline docs/excel/show-hide-add-in.md]: What about content add-ins?

File: docs/excel/show-hide-add-in.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+- a task pane
+- a UI-less function file
+- an Excel custom function
+
+Until now, if an add-in had more than one of these parts, then each part ran in its own separate JavaScript runtime, with its own global object and global variables.
+
+Microsoft is now making it possible for add-ins with two or more parts to share a common JavaScript runtime. This new feature enables new preview APIs for hiding the task pane part of an add-in while the add-in remains running and for reopening the task pane later.
+
+> [!IMPORTANT]
```
Comment (mscharlock) [inline docs/excel/show-hide-add-in.md]: It might be worth adding an includes file specific to shared runtime (David might already have one?) so we don't have to update in multiple places

File: docs/excel/show-hide-add-in.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+Office Add-ins can include any of the following parts:
+
+- a task pane
+- a UI-less function file
+- an Excel custom function
+
+Until now, if an add-in had more than one of these parts, then each part ran in its own separate JavaScript runtime, with its own global object and global variables.
+
+Microsoft is now making it possible for add-ins with two or more parts to share a common JavaScript runtime. This new feature enables new preview APIs for hiding the task pane part of an add-in while the add-in remains running and for reopening the task pane later.
```
Comment (mscharlock) [inline docs/excel/show-hide-add-in.md]:
```suggestion
It's possible for add-ins with two or more parts to share a common JavaScript runtime. This shared runtime feature enables new preview APIs for hiding the task pane part of an add-in while the add-in remains running and for reopening the task pane later.
```
I think I would mention that it's called shared runtime before the next ## section

File: docs/excel/show-hide-add-in.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Show and hide the task pane
+
+The new APIs are in the `Office.addin` namespace. To show the task pane, your code calls `Office.addin.showAsTaskpane()`. Office will display in a task pane the page that you assigned to the resource ID (`resid`) for the task pane. This is the `resid` that you assigned to the `<SourceLocation>` of the `<Action xsi:type="ShowTaskpane">` in the manifest. (See [Configure your Office Add-in to use a shared runtime](configure-your-add-in-to-use-a-shared-runtime.md).)
+
+This is an asynchronous method, so code that should not run until it completes should be awaited, either with the `await` keyword or with a `then()` method, depending on which JavaScript syntax you are using. The following assumes that there is an Excel worksheet named **CurrentQuarterSales**. The task pane should be made visible whenever this worksheet is activated. The method `onCurrentQuarter` is a handler for the [Office.Worksheet.onActivated](/javascript/api/excel/excel.worksheet?view=excel-js-preview#onactivated) event which has been registered for the worksheet.
```
Comment (mscharlock) [inline docs/excel/show-hide-add-in.md]:
```suggestion
This is an asynchronous method, so code that shouldn't run until it completes should be awaited, either with the `await` keyword or with a `then()` method, depending on which JavaScript syntax you are using. The following sample assumes that there is an Excel worksheet named **CurrentQuarterSales**. The task pane should be made visible whenever this worksheet is activated. The method `onCurrentQuarter` is a handler for the [Office.Worksheet.onActivated](/javascript/api/excel/excel.worksheet?view=excel-js-preview#onactivated) event which has been registered for the worksheet.
```
"should be awaited" & "should be made visible" are passive, consider rewriting to active voice

File: docs/excel/show-hide-add-in.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+function onCurrentQuarterDeactivated() {
+    Office.addin.hide();
+}
+```
+
+### Preservation of state and event listeners
+
+The `hide()` and `showAsTaskpane()` methods only change the *visibility* of the task pane. They do not unload or reload it (or reinitialize its state).
+
+Consider the following scenario: A task pane is designed with tabs. The **Home** tab is open when the add-in is first launched. Suppose a user opens the **Settings** tab and, later, code in the task pane calls `hide()` in response to some event. Still later code calls `showAsTaskpane()` in response to another event. The task pane will reappear, and the **Settings** tab is still elected.
```
Comment (mscharlock) [inline docs/excel/show-hide-add-in.md]: Is there a way to illustrate the two scenarios through pictures? Or simplify the written descriptions? I get what they are saying but took me a couple times of moving my hands around to simulate the tabs to figure out what was going on. Not sure if this is specific to me as a physical/visual learner. I'm gonna also tag @AlexJerabek to see what he thinks on this one as I am dealing with some cognitive effects from a condition I have today and unsure if that is coming into play in my difficulty figuring this out quickly.

Comment (mscharlock) [reply docs/excel/show-hide-add-in.md]: I almost don't know if you need the example given you say that it doesn't reload or change state in this first one

File: docs/excel/show-hide-add-in.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+### Handle visibility changed event
+
+Changing the visibility of the task pane is an event that your code can handle. To register a handler for the event, you do not use an "add handler" method as you would in most Office JavaScript contexts. Instead, there is a special function to which you pass your handler: `Office.addin.onVisibilityModeChanged`. The function returns another function that deregisters the handler. The following is an example. Note that the `visibilityMode` property of the `args` object that Office passes to the handler can have the values "Hidden" or "Taskpane".
```
Comment (mscharlock) [inline docs/excel/show-hide-add-in.md]:
```suggestion
You can change the visibility of a task pane through events. To register a handler for the event, don't use an "add handler" method. Instead, pass a special function to your handler: `Office.addin.onVisibilityModeChanged`. The function returns another function that deregisters the handler. 

In the following example, note that the `visibilityMode` property of the `args` object that Office passes to the handler can have the values "Hidden" or "Taskpane".
```

File: docs/excel/show-hide-add-in.md (hunk 1 above)
Comment (Rick-Kirkham) [reply docs/excel/show-hide-add-in.md]: Content add-ins are not yet supported in shared runtime.

File: docs/excel/show-hide-add-in.md (hunk 6 above)
Comment (Rick-Kirkham) [reply docs/excel/show-hide-add-in.md]: Unfortunately, this suggestion changes the meaning. We don't want to say that the visibility changes in response to [unspecified] events. We want to say that a change in the visibility of the add-in (caused by the API, not an event) is itself an event that can be handled.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[3859336](https://github.com/OfficeDev/office-js-docs-pr/commits/3859336e02efc6f9eeab2f28d814c2f4bafeaa8b)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C27%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002272005271344-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=HX7aL8yu10%2B4Ye5Ed%2FLt4LQgb47GoG1X1%2BR0PKqfch8%3D&st=2020-02-27T20%3A02%3A09Z&se=2020-03-29T20%3A07%3A09Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/show-hide-add-in.md (hunk 2 above)
Comment (Rick-Kirkham) [reply docs/excel/show-hide-add-in.md]: Good idea. Done.

File: docs/excel/show-hide-add-in.md (hunk 4 above)
Comment (Rick-Kirkham) [reply docs/excel/show-hide-add-in.md]: done

File: docs/excel/show-hide-add-in.md (hunk 5 above)
Comment (Rick-Kirkham) [reply docs/excel/show-hide-add-in.md]: Good catch. There was a graphic in the private preview Word doc, but I seem to have lost it when converting it to markdown. I've put it back in.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[39710b7](https://github.com/OfficeDev/office-js-docs-pr/commits/39710b782862161a5178bea40c0d51396e7a74ad)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/images/TaskpaneWithTabs.png](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/images/TaskpaneWithTabs.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C27%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002272026099944-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=t%2FgzknctiSfet9NxB5oxLA5nRupAR7xIohNh%2BaMzf2s%3D&st=2020-02-27T20%3A22%3A58Z&se=2020-03-29T20%3A27%3A58Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[f143047](https://github.com/OfficeDev/office-js-docs-pr/commits/f1430472942d7ef59859e4a252283d2ffb7710eb)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/images/TaskpaneWithTabs.png](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/images/TaskpaneWithTabs.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C27%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002272029232797-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=jMCh%2Bn6aH7AGfg%2BQYTwnfXtx79S%2FBd4gA%2FrVmA1AaHs%3D&st=2020-02-27T20%3A26%3A11Z&se=2020-03-29T20%3A31%3A11Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/toc.yml [hunk 7]
```diff
@@ -607,6 +607,9 @@
     - name: Shapes
       href: excel/excel-add-ins-shapes.md
       displayName: Excel
+    - name: Show or hide an Office Add-in in a shared runtime
+      href: excel/show-hide-add-in.md
+      displayName: Excel
```
Comment (AlexJerabek) [inline docs/toc.yml]: Why is this in the Excel section (same goes for the **Run code on document open** article)? It seems like even if they currently only work on Excel, they belong in the **Core Concepts** realm.

File: docs/excel/show-hide-add-in.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+localization_priority: Normal
+---
+
+# Show or hide an Office Add-in in a shared runtime (preview)
+
+Office Add-ins can include any of the following parts:
+
+- a task pane
+- a UI-less function file
+- an Excel custom function
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: I think each item in a list needs to be capitalized. The style guide doesn't explicitly state that, but every example is that way.

File: docs/excel/show-hide-add-in.md (hunk 1 above)
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: Something about the plural sounds weird to me. Consider the following:
```suggestion
An Office Add-in can include any of the following parts:
```

File: docs/excel/show-hide-add-in.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+Until now, if an add-in had more than one of these parts, then each part ran in its own separate JavaScript runtime, with its own global object and global variables.
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: "Until now" will get dated fast. Can we reference the requirement set instead?

File: docs/excel/show-hide-add-in.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+It's possible for add-ins with two or more parts to share a common JavaScript runtime. This shared runtime feature enables new preview APIs for hiding the task pane part of an add-in while the add-in remains running and for reopening the task pane later.
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: Suggestion for conciseness and clarity:
```suggestion
It's possible for add-ins with two or more parts to share a common JavaScript runtime. This shared runtime feature enables new preview APIs that hide and reopen the task pane while the add-in runs.
```

File: docs/excel/show-hide-add-in.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+> [!INCLUDE [Information about using preview APIs](../includes/excel-shared-runtime-preview-note.md)]
+
+## Configure an add-in to use a shared runtime
+
+To configure the add-in to use a shared runtime, see [Configure your Office Add-in to use a shared runtime](configure-your-add-in-to-use-a-shared-runtime.md).
+
+## Show and hide the task pane
+
+The new APIs are in the `Office.addin` namespace. To show the task pane, your code calls `Office.addin.showAsTaskpane()`. Office will display in a task pane the page that you assigned to the resource ID (`resid`) for the task pane. This is the `resid` that you assigned to the `<SourceLocation>` of the `<Action xsi:type="ShowTaskpane">` in the manifest. (See [Configure your Office Add-in to use a shared runtime](configure-your-add-in-to-use-a-shared-runtime.md).)
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: `Office.addin` isn't technically a namespace, right? It's a read-only property, like `context`.

Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: I'm not sure you need to link the **Configure** article again. It's reference in the previous paragraph.

File: docs/excel/show-hide-add-in.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+
+This is an asynchronous method, so it your code should await it when the code that follows it should not run until it completes, either with the `await` keyword or with a `then()` method, depending on which JavaScript syntax you are using. The following assumes that there is an Excel worksheet named **CurrentQuarterSales**. The add-in should make the task pane visible whenever this worksheet is activated. The method `onCurrentQuarter` is a handler for the [Office.Worksheet.onActivated](/javascript/api/excel/excel.worksheet?view=excel-js-preview#onactivated) event which has been registered for the worksheet.
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: Typo? Also, small grammar suggestion.
```suggestion
This is an asynchronous method, so your code should await it when the subsequent code should not run until it completes. Wait for this completion with either the `await` keyword or a `then()` method, depending on which JavaScript syntax you are using. The following assumes that there is an Excel worksheet named **CurrentQuarterSales**. The add-in should make the task pane visible whenever this worksheet is activated. The method `onCurrentQuarter` is a handler for the [Office.Worksheet.onActivated](/javascript/api/excel/excel.worksheet?view=excel-js-preview#onactivated) event which has been registered for the worksheet.
```

File: docs/excel/show-hide-add-in.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+function onCurrentQuarterDeactivated() {
+    Office.addin.hide();
+}
+```
+
+### Preservation of state and event listeners
+
+The `hide()` and `showAsTaskpane()` methods only change the *visibility* of the task pane. They do not unload or reload it (or reinitialize its state).
+
+Consider the following scenario: A task pane is designed with tabs. The **Home** tab is open when the add-in is first launched. Suppose a user opens the **Settings** tab and, later, code in the task pane calls `hide()` in response to some event. Still later code calls `showAsTaskpane()` in response to another event. The task pane will reappear, and the **Settings** tab is still elected.
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: Typo?
```suggestion
Consider the following scenario: A task pane is designed with tabs. The **Home** tab is open when the add-in is first launched. Suppose a user opens the **Settings** tab and, later, code in the task pane calls `hide()` in response to some event. Still later code calls `showAsTaskpane()` in response to another event. The task pane will reappear, and the **Settings** tab is still selected.
```

File: docs/excel/show-hide-add-in.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+![A screenshot of task pane that has four tabs labelled Home, Settings, Favorites, and Accounts.](../images/TaskpaneWithTabs.png)
+
+In addition, any event listeners that are registered in the task pane continue to run even when the task pane is hidden.
+
+Consider the following scenario: The task pane has a registered handler for the Excel `Worksheet.onActivated` and `Excel.Worksheet.onDeactivated` events for a sheet named **Sheet1**. The activated handler causes a green dot to appear in the task pane. The deactivated handler turns the dot red (which is its default state). Suppose then that code calls `hide()` when **Sheet1** is not activated and the dot is red. While the task pane is hidden, **Sheet1** is activated. Later code calls `showAsTaskpane()` in response to some event. When the task pane opens, the dot is green because the event listeners and handlers ran even though the task pane was hidden.
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: In this line, sometimes the `Excel` namespace is used and sometimes it's not. We should be consistent. I suggest omitting it.

File: docs/excel/show-hide-add-in.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+### Handle visibility changed event
+
+Changing the visibility of the task pane is an event that your code can handle. To register a handler for the event, you do not use an "add handler" method as you would in most Office JavaScript contexts. Instead, there is a special function to which you pass your handler: `Office.addin.onVisibilityModeChanged`. The function returns another function that deregisters the handler. The following is an example. Note that the `visibilityMode` property of the `args` object that Office passes to the handler can have the values "Hidden" or "Taskpane".
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: I also find this example hard to parse. Is it possible to rewrite this as a bit of stubbed-out code?

File: docs/excel/show-hide-add-in.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+var removeVisibilityModeHandler =
+    await Office.addin.onVisibilityModeChanged(function(args) {
+        console.log('Visibility is changed to ' + args.visibilityMode);
+});
+```
+
+To deregister the handler:
+
+```javascript
+await removeVisibilityModeHandler();
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: Are we officially okay with having `await` in the JS examples? If so, maybe we should consider cleaning up some older `then()` examples.

File: docs/toc.yml (hunk 7 above)
Comment (Rick-Kirkham) [reply docs/toc.yml]: I'm following the lead of @davidchesnut in the placement in the TOC for all the shared runtime-related docs. I'll leave it where it is for now and confer with him.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[50f4e8a](https://github.com/OfficeDev/office-js-docs-pr/commits/50f4e8a4ca97ed080aae7195108e56be0dd0df40)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/images/TaskpaneWithTabs.png](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/images/TaskpaneWithTabs.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C27%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002272155083150-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=ZRR6PNlqq1jZFM%2FEcBKGsJwcEtwQbppVsVwWgKnMZqQ%3D&st=2020-02-27T21%3A51%3A41Z&se=2020-03-29T21%3A56%3A41Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/show-hide-add-in.md (hunk 15 above)
Comment (Rick-Kirkham) [reply docs/excel/show-hide-add-in.md]: rewrote the whole section

File: docs/excel/show-hide-add-in.md (hunk 16 above)
Comment (Rick-Kirkham) [reply docs/excel/show-hide-add-in.md]: rewrote whole section

**Comment by o365devx** [comment]
Docs Build status updates of commit _[6694606](https://github.com/OfficeDev/office-js-docs-pr/commits/6694606b35bc573d0daffce07165302cd52bd132)_: 

:clock10: **Pending**: waiting for processors ([8](https://ops.microsoft.com/#/statistic) builds ahead of you) 

 :warning: **Docs Build is busy**, currently there are 8 builds ahead of this one, for more information you can view the Build queue graph on the [Docs Portal](https://ops.microsoft.com/#/statistic).

**Comment by o365devx** [comment]
Docs Build status updates of commit _[6694606](https://github.com/OfficeDev/office-js-docs-pr/commits/6694606b35bc573d0daffce07165302cd52bd132)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/images/TaskpaneWithTabs.png](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/images/TaskpaneWithTabs.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002280129186955-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=NWRyd1ILGIBQpZF%2F%2BdjElzewBQvGgi0ZwFeH2XmAN%2Fc%3D&st=2020-02-28T01%3A25%3A52Z&se=2020-03-30T01%3A30%3A52Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/excel/show-hide-add-in.md [hunk 17]
```diff
@@ -51,27 +51,83 @@ function onCurrentQuarterDeactivated() {
...
+When your code changes the visibility of the task pane with `showAsTaskpane()` or `hide()`, Office triggers the VisibilityModeChanged event. It can be useful to handle this event. For example, suppose the task pane displays a list of all the sheets in a workbook. If a new worksheet is added while the task pane is hidden, making the task pane visible would not, in itself, add the new worksheet name to the list. But your code can respond to the VisibilityModeChanged event to reload the [Worksheet.name](/javascript/api/excel/excel.worksheet#name) property of all the worksheets in the [Workbook.worksheets](/javascript/api/excel/excel.workbook#worksheets) collection as shown in the example code below.
```
Comment (AlexJerabek) [inline docs/excel/show-hide-add-in.md]: Missing code ticks around the event name.
```suggestion
When your code changes the visibility of the task pane with `showAsTaskpane()` or `hide()`, Office triggers the `VisibilityModeChanged` event. It can be useful to handle this event. For example, suppose the task pane displays a list of all the sheets in a workbook. If a new worksheet is added while the task pane is hidden, making the task pane visible would not, in itself, add the new worksheet name to the list. But your code can respond to the `VisibilityModeChanged` event to reload the [Worksheet.name](/javascript/api/excel/excel.worksheet#name) property of all the worksheets in the [Workbook.worksheets](/javascript/api/excel/excel.workbook#worksheets) collection as shown in the example code below.
```

**Comment by o365devx** [comment]
Docs Build status updates of commit _[dd33364](https://github.com/OfficeDev/office-js-docs-pr/commits/dd33364bce9e7d350ba2c8004b79c6f500b464c1)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/images/TaskpaneWithTabs.png](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/images/TaskpaneWithTabs.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002281810026693-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=UZ5GUmytN%2BvF52%2F3QFIiGH7kJelha9GwWaHUUbeZS8M%3D&st=2020-02-28T18%3A06%3A31Z&se=2020-03-30T18%3A11%3A31Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[2ee79cb](https://github.com/OfficeDev/office-js-docs-pr/commits/2ee79cb42daeb70320cce851d001ba685c03b643)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/show-hide-add-in.md](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/excel/show-hide-add-in.md) | :white_check_mark:Succeeded | |
[docs/images/TaskpaneWithTabs.png](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/images/TaskpaneWithTabs.png) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/show-hide-taskpane/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C2%5C28%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202002281813589782-1623%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=kLBA7lYzbFoZQ6aOd%2BAkFPkdTfwQvVyg5x9nwtnG%2BmY%3D&st=2020-02-28T18%3A10%3A31Z&se=2020-03-30T18%3A15%3A31Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by lindalu-MSFT** [comment]
Hi @Rick-Kirkham sorry I didn't get a chance to review but it looks like you had two reviewers, so you're probably fine! LMK if you want me to still have a look. Thanks, lindalu
